x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/aggregations/metrics/AnalyticsAggregatorFactory.java

```diff
@@ -65,6 +68,24 @@ public static void registerPercentileRanksAggregator(ValuesSourceRegistry.Builde
     public static void registerHistoBackedSumAggregator(ValuesSourceRegistry.Builder builder) {
         builder.register(SumAggregationBuilder.NAME,
             AnalyticsValuesSourceType.HISTOGRAM,
-            (MetricAggregatorSupplier) HistoBackedSumAggregator::new);
+            (MetricAggregatorSupplier) (name, valuesSource, format, context, parent, metadata) ->
+                new HistoBackedSumAggregator(name, (HistogramValuesSource.Histogram) valuesSource, format, context, parent, metadata)
```

Comment:
`(MetricAggregatorSupplier) HistoBackedSumAggregator::new` is definitely better looking and I would prefer it as well. 

The problem is that `MetricAggregatorSupplier` passes a `ValuesSource` argument
```
Aggregator build(String name,
                     ValuesSource valuesSource,
                     DocValueFormat format,
                     SearchContext context,
                     Aggregator parent,
                     Map<String, Object> metadata) throws IOException;
```
while `HistoBackedSumAggregator` constructor expects a `HistogramValuesSource.Histogram`
```
HistoBackedSumAggregator(String name, HistogramValuesSource.Histogram valuesSource, DocValueFormat formatter, SearchContext context,
            Aggregator parent, Map<String, Object> metadata) throws IOException 
```

So I had 3 options:

1. Keep the `ValuesSource` parameter in the `HistoBackedSumAggregator` constructor and do a type cast later when I would use the `HistogramValuesSource.Histogram` object.
2. Create a separate `HistoBackedMetricAggregatorSupplier` that would match the arguments of the `HistoBackedSumAggregator`
3. Do the cast of `ValuesSource` to `HistogramValuesSource.Histogram` in the registrar method changing the admittedly prettier `(MetricAggregatorSupplier) HistoBackedSumAggregator::new` to the longer lambda expression.

I chose the third option because it seemed cleaner to do the type cast at the registrar method where it is obvious that the VS is a histogram. On the other hand, I didn't  want to create yet another `AggregatorSupplier` sub-class